config/routes.rb

```diff
@@ -0,0 +1,108 @@
+Rails.application.routes.draw do
+
+  root 'users#index'
+
+  get  '/users/all_markets'  => 'users#all_markets', as: "all_markets"
```

Comment:
Just generally speaking, it might be beneficial on future projects to stick with a more traditional routing and view scheme. Here is a Rails resource on what I mean by this: http://guides.rubyonrails.org/routing.html